pkg/git/git_test.go

```diff
@@ -1,23 +1,271 @@
 package git
 
 import (
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"reflect"
 	"testing"
+
+	"github.com/arschles/assert"
 )
 
-func TestRemoteURL(t *testing.T) {
+func TestRepositoryURL(t *testing.T) {
 	t.Parallel()
 
-	actual := RemoteURL("deis.example.com", "app")
-	expected := "ssh://git@deis-builder.example.com:2222/app.git"
+	actual := RepositoryURL("deis.example.com", "app")
+	assert.Equal(t, actual, "ssh://git@deis-builder.example.com:2222/app.git", "url")
+	actual = RepositoryURL("deis.10.245.1.3.xip.io:31350", "velcro-underdog")
+	assert.Equal(t, actual, "ssh://git@deis-builder.10.245.1.3.xip.io:2222/velcro-underdog.git", "url")
+}
+
+func TestGetRemotes(t *testing.T) {
+	t.Parallel()
 
-	if actual != expected {
-		t.Errorf("Expected %s, Got %s", expected, actual)
+	expected := []remote{
+		{"test", "ssh://test.com/test.git"},
+		{"example", "ssh://example.com:2222/example.git"},
 	}
 
-	actual = RemoteURL("deis.10.245.1.3.xip.io:31350", "velcro-underdog")
-	expected = "ssh://git@deis-builder.10.245.1.3.xip.io:2222/velcro-underdog.git"
+	actual, err := getRemotes(func(cmd []string) (string, error) {
+		assert.Equal(t, cmd, []string{"remote", "-v"}, "args")
+		return `test	ssh://test.com/test.git (fetch)
```

Comment:
github is odd. there is a tab between "test" and "ssh" here. But github doesn't seem to show it!